pkg/build/controller/build_controller.go

```diff
@@ -130,6 +132,16 @@ func (bc *BuildController) HandleBuild(build *buildapi.Build) error {
 		return err
 	}
 
+	// Handle processing ValueFrom in build strategy environment variables
+	originalEnvVars := buildutil.GetEnvFromStrategy(build)
+	processedEnvVars, err := common.ProcessValueFrom(build, bc.KubeClient)
+	if err != nil {
+		build.Status.Phase = buildapi.BuildPhaseFailed
+		build.Status.Reason = buildapi.StatusReasonInvalidEnvironmentVariable
+		return err
+	}
+	*originalEnvVars = *processedEnvVars
```

Comment:
I think i had a comment elsewhere that the processing of the env vars should be done in the same place/way that we're doing it for the builddefaulter/overrider.